### PR TITLE
Improvised bone fixes no longer take a ridiculous amount of time to heal

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
After a hectic round that involved quite a lot of people being smashed by vending machines and breaking their bones, a medical player mentioned to me that, while they appreciated that I put in an improvised treatment for broken bones for doctors dealing with a massive flood of bodies, it was very spotty and seemed to take an awful long time or outright not work at all. Confused, I broke someone's legs and applied the improvised fix, and found that the times for recovery were incredibly long, 10 minutes for a severe bone wound, and an astounding 20 minutes for a critical bone wound!

As it turns out, I made the classic mistake of mixing seconds and ticks, since mob life ticks generally only happen every 2 seconds, but I did my math using seconds, making everything take twice as long (probably more if it was laggy!) as it was supposed to. Oops!

So this cuts the ticks needed for the wound to heal in half, lowers them a bit more for good measure, and makes so sleeping or lying down helps speed the process. It now takes about 4 minutes for the improvised bone gel + surgical tape fix to heal a severe bone wound, and about 8 minutes for it to heal a critical bone wound. This recovery is about 50% faster while you're asleep, or about 25% faster if you're just laying down. Applying it to yourself still makes it take 50% longer to heal, and no matter what you'll still suffer random spikes of damage throughout the recovery (a bit more often than before to keep surgery the better option).

## Why It's Good For The Game
Removes an accidental noob trap for doctors and people trying to fix themselves, as well as a reason to sleep or lay down while badly wounded and recovering.

## Changelog
:cl: Ryll/Shaps
fix: Improvised bone wound treatments with bone gel and surgical tape applied outside of surgery no longer take ridiculous amounts of time, down from 10/20 minutes for severe/critical bone wounds to 4/8 minutes.
balance: Sleeping while an improvised bone wound treatment is processing will cause it to heal about 50% faster, while laying down while awake will heal it about 25% faster. Your bones will spike with pain a bit more frequently during the healing though.
/:cl:

### CODE REVIEW

File: code/datums/wounds/bones.dm [hunk 1]
```diff
@@ -65,16 +65,22 @@
 			active_trauma = victim.gain_trauma_type(brain_trauma_group, TRAUMA_RESILIENCE_WOUND)
 		next_trauma_cycle = world.time + (rand(100-WOUND_BONE_HEAD_TIME_VARIANCE, 100+WOUND_BONE_HEAD_TIME_VARIANCE) * 0.01 * trauma_cycle_cooldown)
 
-	if(!regen_points_needed)
+	if(!(gelled && taped))
```
Comment (Rohesie) [inline code/datums/wounds/bones.dm:68]:
```suggestion
	if(!gelled || !taped)
```

File: code/datums/wounds/bones.dm [hunk 2]
```diff
@@ -65,16 +65,22 @@
...
 		return
 
-	regen_points_current++
-	if(prob(severity * 2))
+	regen_ticks_current++
+	if(victim.IsSleeping() && prob(50))
+		regen_ticks_current++
+	else if(!(victim.mobility_flags & MOBILITY_STAND) && prob(50))
+		regen_ticks_current += 0.5
```
Comment (Rohesie) [inline code/datums/wounds/bones.dm:75]:
```suggestion
	if(victim.body_position == LYING_DOWN)
		if(prob(50))
			regen_ticks_current += 0.5
		if(victim.IsSleeping() && prob(50))
			regen_ticks_current += 0.5
```

**Comment by Jared-Fogle** [comment]
aren't we supposed to be using delta times for process now instead of thinking in terms of ticks?

File: code/datums/wounds/bones.dm [hunk 3]
```diff
@@ -65,16 +65,22 @@
 			active_trauma = victim.gain_trauma_type(brain_trauma_group, TRAUMA_RESILIENCE_WOUND)
 		next_trauma_cycle = world.time + (rand(100-WOUND_BONE_HEAD_TIME_VARIANCE, 100+WOUND_BONE_HEAD_TIME_VARIANCE) * 0.01 * trauma_cycle_cooldown)
 
-	if(!regen_points_needed)
+	if(!gelled || !taped))
```
Comment (Rohesie) [inline code/datums/wounds/bones.dm:68]:
```suggestion
	if(!gelled || !taped)
```

**Review by kriskog — APPROVED** [review]

**Pull request merged**